src/components/StakingVaults/ProviderCard.tsx

```diff
@@ -64,19 +65,23 @@ export const ProviderCard: React.FC<ProviderCardProps> = ({
 
   const isSnapInstalled = useIsSnapInstalled()
 
-  const filteredDownStakingOpportunities = stakingOpportunities.filter(
-    e =>
+  const filteredDownStakingOpportunities = stakingOpportunities.filter(e => {
+    const chainAccountIds = selectAccountIdsByChainId(store.getState(), { chainId: e.chainId })
+    return (
       staking.includes(e.id as OpportunityId) &&
-      walletSupportsChain({ chainId: e.chainId, wallet, isSnapInstalled }),
-  )
+      walletSupportsChain({ chainId: e.chainId, wallet, isSnapInstalled, chainAccountIds })
+    )
+  })
 
   const lpOpportunities = useAppSelector(selectAggregatedEarnUserLpOpportunities)
 
-  const filteredDownLpOpportunities = lpOpportunities.filter(
-    e =>
+  const filteredDownLpOpportunities = lpOpportunities.filter(e => {
```

Comment:
this looks like something that should be memoized

Reply:
08f3cb107b4ba50ed0f06d50b8b0f903b628b4a3